dipy/reconst/ivim.py

```diff
@@ -0,0 +1,478 @@
+
+    Parameters
+    ----------
+    params : array
+        The value of f and D_star.
+
+    gtab : GradientTable class instance
+        Gradient directions and bvalues.
+
+    signal : array
+        Array containing the actual signal values.
+
+    other_params : array
+        The parameters S0 and D which are fixed and obtained from a linear fit.
+    """
+    f, D_star = params
+    S0, D = x0[0], x0[3]
+
+    return signal - f_D_star_prediction([f, D_star], gtab, x0)
+
+
+class IvimModel(ReconstModel):
+    """Ivim model
+    """
+
+    def __init__(self, gtab, split_b_D=400.0, split_b_S0=200., bounds=None,
+                 tol=1e-15, scaling=np.array([10e-4, 10e2, 1000., 10000.]),
+                 options={'gtol': 1e-15, 'ftol': 1e-15,
+                          'eps': 1e-15, 'maxiter': 1000}):
+        """
+        Initialize an IVIM model.
+
+        The IVIM model assumes that biological tissue includes a volume
+        fraction 'f' of water flowing in perfused capillaries, with a
+        perfusion coefficient D* and a fraction (1-f) of static (diffusion
+        only), intra and extracellular water, with a diffusion coefficient
+        D. In this model the echo attenuation of a signal in a single voxel
+        can be written as
+
+            .. math::
+
+            S(b) = S_0[f*e^{(-b*D\*)} + (1-f)e^{(-b*D)}]
+
+            Where:
+            .. math::
+
+            S_0, f, D\* and D are the IVIM parameters.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+            Gradient directions and bvalues
+
+        split_b_D : float, optional
+            The b-value to split the data on for two-stage fit. This will be
+            used while estimating the value of D. The assumption is that at
+            higher b values the effects of perfusion is less and hence the
+            signal can be approximated as a mono-exponenetial decay.
+            default : 400.
+
+        split_b_S0 : float, optional
+            The b-value to split the data on for two-stage fit for estimation
+            of S0 and initial guess for D_star. The assumption here is that
+            at low bvalues the effects of perfusion are more.
+            default : 200.
+
+        bounds : tuple of arrays with 4 elements, optional
+            Bounds to constrain the fitted model parameters. This is only
+            supported for Scipy version > 0.17. When using a older scipy
+            version, this function will raise an error if bounds are different
+            from None. default : ([0., 0., 0., 0.], [np.inf, 1., 1., 1.])
+
+        tol : float, optional
+            Tolerance for convergence of minimization.
+            default : 1e-7
+
+        scaling : array, optional
+            Scaling for the parameters. This is passed to `_ivim_error` to
+            normalize the signal and scale the parameters appropriately.
+            default: [1e-4, 100., 1000., 10000.]
```

Comment:
A bit more guidance would be good. For example, something like: "these scaling factors should bring the different parameters more or less to the same order of magnitude, so should be inversely proportional to the scale of S0, f, D D_star and D, respectively."